File: Solutions/han3l/Game Shop/Part1/Item.cpp
#include "Item.h"

Item::Item() 
{
}

Item::~Item() 
{
}


Item::Item(std::string name, std::string description, int weight, int value) 
	: name(name), description(description), weight(weight), value(value) 
{
}

void Item::Describe()
{
	std::cout << "Name\t\t= " << this->name << std::endl;
	std::cout << "Description\t= " << this->description << std::endl;
	std::cout << "Weight\t\t= " << this->weight << " lbs" << std::endl;
	std::cout << "Value\t\t= " << this->value << " gold coins" << std::endl << std::endl;
}

File: Solutions/DSMdongly/game-shop/part-1/src/game-shop/item/item.go
package item

import (
	"fmt"
)

type Item struct {
	Name        string
	Description string
	Weight      int
	Value       int
}

func NewItem(nme string, dsc string, wgt int, val int) *Item {
	return &Item{
		Name:        nme,
		Description: dsc,
		Weight:      wgt,
		Value:       val,
	}
}

func (ite Item) Describe() {
	fmt.Println()
	fmt.Printf("Name        = %s\n", ite.Name)
	fmt.Printf("Description = %s\n", ite.Description)
	fmt.Printf("Weight      = %d lbs\n", ite.Weight)
	fmt.Printf("Value       = %d gold coins\n", ite.Value)
}
